src/network/prover.rs

```diff
@@ -272,4 +347,85 @@ impl<N: Network, E: Environment> Prover<N, E> {
+                            let state = prover.state.clone();
+                            let thread_pool = prover.thread_pool.clone();
+                            let canon = prover.ledger_reader.clone(); // This is *safe* as the ledger only reads.
+                            let unconfirmed_transactions = prover.memory_pool.read().await.transactions();
+                            let ledger_router = prover.ledger_router.clone();
+                            let prover_router = prover.prover_router.clone();
+
+                            tasks_clone.append(task::spawn(async move {
+                                // Mine the next block.
+                                let result = task::spawn_blocking(move || {
+                                    thread_pool.install(move || {
+                                        canon.mine_next_block(
+                                            recipient,
+                                            E::COINBASE_IS_PUBLIC,
+                                            &unconfirmed_transactions,
+                                            &terminator,
+                                            &mut thread_rng(),
+                                        )
+                                    })
+                                })
+                                .await
+                                .map_err(|e| e.into());
+
+                                // Set the status to `Ready`.
+                                status.update(State::Ready);
+
+                                match result {
+                                    Ok(Ok((block, coinbase_record))) => {
+                                        debug!("Miner has found unconfirmed block {} ({})", block.height(), block.hash());
+                                        // Store the coinbase record.
+                                        if let Err(error) = state.add_coinbase_record(block.height(), coinbase_record) {
+                                            warn!("[Miner] Failed to store coinbase record - {}", error);
+                                        }
+
+                                        // Broadcast the next block.
+                                        let request = LedgerRequest::UnconfirmedBlock(local_ip, block, prover_router.clone());
+                                        if let Err(error) = ledger_router.send(request).await {
+                                            warn!("Failed to broadcast mined block - {}", error);
+                                        }
+                                    }
+                                    Ok(Err(error)) | Err(error) => trace!("{}", error),
+                                }
+                            }));
+                        }
+                        // Sleep for 2 seconds.
+                        tokio::time::sleep(std::time::Duration::from_secs(2)).await;
```

Comment:
nit: could probably be `const`ified for greater visibility; could be done in a follow-up